chia/cmds/init_funcs.py

```diff
@@ -107,6 +108,38 @@ def check_keys(new_root: Path) -> None:
     save_config(new_root, "config.yaml", config)
 
 
+#
+# Checks if the given key is used for either the farmer rewards or pool rewards
+# returns a tuple of two booleans
+# The first is true if the key is used as the Farmer rewards, otherwise false
+# The second is true if the key is used as the Pool rewards, otherwise false
+# Returns both false if the key cannot be found with the given fingerprint
+#
+def check_key_used_for_rewards(new_root: Path, sk: PrivateKey, max_ph_to_search: int) -> Tuple[bool, bool]:
+    if sk is None:
+        return False, False
+
+    config: Dict = load_config(new_root, "config.yaml")
+    farmer_target = config["farmer"].get("xch_target_address")
+    pool_target = config["pool"].get("xch_target_address")
```

Comment:
We should also check for any keys used for pooling rewards. For that, we need to look at the config file:
https://github.com/Chia-Network/chia-blockchain/blob/pools.dev/chia/farmer/farmer.py#L450

This is only in pools.dev branch though, not main yet